src/console/MeasurementCollectionToFhir/Processor.cs

```diff
@@ -34,15 +38,50 @@ public class Processor : IEventConsumer
             _templateManager = templateManager;
             _measurementImportService = measurementImportService;
             _logger = logger;
+            _retryPolicy = CreateRetryPolicy(logger);
         }
 
         public async Task ConsumeAsync(IEnumerable<IEventMessage> events)
         {
             EnsureArg.IsNotNull(events);
             EnsureArg.IsNotNull(_templateDefinition);
 
-            var templateContent = _templateManager.GetTemplateAsString(_templateDefinition);
+            await _retryPolicy.ExecuteAsync(async () => await ConsumeAsyncImpl(events, _templateManager.GetTemplateAsString(_templateDefinition)));
+        }
+
+        private async Task ConsumeAsyncImpl(IEnumerable<IEventMessage> events, string templateContent)
+        {
             await _measurementImportService.ProcessEventsAsync(events, templateContent, _logger).ConfigureAwait(false);
         }
+
+        private static AsyncPolicy CreateRetryPolicy(ITelemetryLogger logger)
+        {
+            bool ExceptionRetryableFilter(Exception ee)
+            {
+                logger.LogError(new Exception("Encountered retryable exception", ee));
+                TrackExceptionMetric(ee, logger);
+                return true;
+            }
+
+            return Policy
+                .Handle<Exception>(ExceptionRetryableFilter)
+                .WaitAndRetryForeverAsync(retryCount => TimeSpan.FromSeconds(Math.Min(30, Math.Pow(2, retryCount))));
```

Comment:
>30 [](http://example.com/codeflow?start=86&length=2)

30 seconds may be too long for an initial retry. Would suggest 5 to 10.

Reply:
The time between retries will be the lower of 30 seconds and the Math.Pow function. Since the Pow function is based on the retry count, the initial retry should be 2 ^ 1 = 2 seconds.

Reply:
Or do you mean that we should lower the maximum time between all retries to 10 seconds from 30?

Reply:
I misread the code line.  Thanks for the clarification.  I think we are good here.